### PR TITLE
Add cluster scanner metrics in metrics-v3

### PR DESCRIPTION
**Pull request by anjalshireesh**
## Community Contribution License

All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

endpoint: /minio/metrics/v3/cluster/scanner
metrics:
 - bucket_scans_finished (counter)
 - bucket_scans_started (counter)
 - directories_scanned (counter)
 - last_activity_nano_seconds (gauge)
 - objects_scanned (counter)
 - versions_scanned (counter)

## Motivation and Context

https://github.com/miniohq/engineering/issues/1510

## How to test this PR?

`curl "http(s)://<host:port>/minio/metrics/v3/cluster/scanner"`

should return the new cluster scanner related metrics

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by balamurugana — APPROVED** [review]

File: cmd/metrics-v3-cluster-scanner.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"time"
+)
+
+const (
+	scannerBucketScansFinished     = "bucket_scans_finished"
+	scannerBucketScansStarted      = "bucket_scans_started"
+	scannerDirectoriesScanned      = "directories_scanned"
+	scannerObjectsScanned          = "objects_scanned"
+	scannerVersionsScanned         = "versions_scanned"
+	scannerLastActivityNanoSeconds = "last_activity_nano_seconds"
```
Comment (vadmeste) [inline cmd/metrics-v3-cluster-scanner.go]: if this is not  a breaking change, let's avoid the nano seconds and do second instead (nano is an overkill and not suitable for human consumption)

Comment (anjalshireesh) [reply cmd/metrics-v3-cluster-scanner.go]: @vadmeste I don't think this is a breaking change as it's (v3) a fresh implementation of the metrics. So I've changed it to seconds as you suggested. PTAL.

File: cmd/metrics-v3-cluster-scanner.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	scannerObjectsScanned      = "objects_scanned"
+	scannerVersionsScanned     = "versions_scanned"
+	scannerLastActivitySeconds = "last_activity_seconds"
+)
+
+var (
+	scannerBucketScansFinishedMD = NewCounterMD(scannerBucketScansFinished,
+		"Total number of bucket scans finished since server start")
+	scannerBucketScansStartedMD = NewCounterMD(scannerBucketScansStarted,
+		"Total number of bucket scans started since server start")
```
Comment (klauspost) [inline cmd/metrics-v3-cluster-scanner.go]: The description is not really representative. It is more like "buckets sets scanned" - but I don't really give any idea... Understanding the value really requires some knowledge of the internals.

Comment (anjalshireesh) [reply cmd/metrics-v3-cluster-scanner.go]: @klauspost I've taken this description from metrics-v2. Let me know if this needs to change.

Comment (klauspost) [reply cmd/metrics-v3-cluster-scanner.go]: Yes. It should probably be changed.

Comment (anjalshireesh) [reply cmd/metrics-v3-cluster-scanner.go]: @klauspost to what? "buckets sets scanned"? Or something else? Sorry I got confused by the statement "but I don't really give any idea"

**Review by shtripat — APPROVED** [review]
lgtm

**Comment by harshavardhana** [comment]
PTAL at the conflicts @anjalshireesh

**Pull request merged**